lib/KMultiSelect.vue

```diff
@@ -0,0 +1,1084 @@
+          class="dropdown-list"
+          role="listbox"
+          aria-multiselectable="true"
+          :style="{
+            margin: '0',
+            padding: '0',
+            listStyle: 'none',
+            outline: 'none',
+          }"
+          :aria-labelledby="ariaLabelledby"
+          :aria-describedby="ariaDescribedById"
+          tabindex="-1"
+        >
+          <!-- Select All Option -->
+          <li
+            v-if="showSelectAll"
+            :id="`select-all-${uid}`"
+            role="option"
+            :class="$computedClass(getSelectAllStyles())"
+            :aria-selected="false"
+            :style="{
+              padding: '8px 12px',
+              cursor: 'pointer',
+              display: 'flex',
+              alignItems: 'center',
+              gap: '8px',
+              minHeight: '40px',
+              borderBottom: `1px solid ${$themeTokens.fineLine}`,
+              fontWeight: 'bold',
+              backgroundColor: isSelectAllFocused ? $themeTokens.primary + '20' : 'transparent',
+              outline: isSelectAllFocused ? `2px solid ${$themeTokens.primary}` : 'none',
+              outlineOffset: isSelectAllFocused ? '-2px' : '0',
+            }"
+            :tabindex="isSelectAllFocused ? 0 : -1"
+            @click.stop="selectAll"
+            @mouseenter="setFocusedSelectAll"
+            @keydown="handleOptionKeydown($event, null, -1)"
+            @focus="setFocusedSelectAll"
+            @mousedown.prevent
+          >
+            <KCheckbox
+              presentational
+              :checked="allOptionsSelected"
+              :indeterminate="someOptionsSelected && !allOptionsSelected"
+              :style="{ flexShrink: 0 }"
+              tabindex="-1"
+              aria-hidden="true"
+            />
+            <span :style="{ flex: 1 }">Select All</span>
+          </li>
+
+          <li
+            v-for="(option, index) in displayedOptions"
+            :id="getElementOptionId(option)"
+            :key="option.id"
+            :ref="el => { if (el) optionRefs[index] = el }"
+            role="option"
+            :class="$computedClass(getOptionStyles(option))"
+            :aria-selected="isOptionSelected(option).toString()"
+            :aria-setsize="displayedOptions.length"
+            :aria-posinset="index + 1"
+            :style="{
+              padding: '8px 12px',
+              cursor: 'pointer',
+              display: 'flex',
+              alignItems: 'center',
+              gap: '8px',
+              minHeight: '40px',
+              backgroundColor: getOptionBackgroundColor(option),
+              outline: getOptionOutline(option),
+              outlineOffset: getOptionOutlineOffset(option),
+            }"
+            :tabindex="focusedOption?.id === option.id ? 0 : -1"
+            @click.stop="toggleOption(option)"
+            @mouseenter="setFocusedOption(option)"
+            @keydown="handleOptionKeydown($event, option, index)"
```

Comment:
Would it be possible to attach `@keydown` listener to the whole list `<ul>` instead of each `<li>` option? This component will be used with large datasets, and adding a listener to every option can hurt performance. Generally, it’s a good web development pattern to use event delegation - attaching a single even listener to the parent.

I used `@keydown` here as one example - it applies to as many events as possible.